src/sentry/api/paginator.py

```diff
@@ -116,10 +126,28 @@ def get_result(self, limit=100, cursor=None):
         return build_cursor(
             results=results,
             limit=limit,
+            hits=hits,
+            max_hits=max_hits,
             cursor=cursor,
             key=self.get_item_key,
         )
 
+    def count_hits(self, queryset):
+        hits_query = self.queryset.values()[:1000].query
+        # clear out any select fields (include select_related) and pull just the id
+        hits_query.clear_select_clause()
+        hits_query.add_fields(['id'])
+        hits_query.clear_ordering(force_empty=True)
+        try:
+            h_sql, h_params = hits_query.sql_with_params()
+        except EmptyResultSet:
+            return 0
+        cursor = connections[self.queryset.db].cursor()
+        cursor.execute('SELECT COUNT(*) FROM ({}) as t'.format(
+            h_sql,
+        ), h_params)
+        return cursor.fetchone()[0]
```

Comment:
Following up on the first comment, this would be a good place to return the struct with hits and the limit, and most of the logic can be removed from `get_result` and encapsulated here instead.